Get my changes.

megawebmaster opened the pull request:
Here you got my little changes in Trinity, which helped me a lot to develop my blog. It is not ended yet, so I think that there will be more changes.

Regards.

zyxist committed:
Removing the custom implementation of event dispatcher in favour of Symfony 2 components.

zyxist committed:
Changing the event dispatcher initialization.

zyxist committed:
Modifying the object locators to support the new event dispatcher.

zyxist committed:
Updating the area interface to fire an event.

zyxist committed:
Changing the event dispatcher.

zyxist committed:
Adding support for custom extensions to the controller manager.

zyxist committed:
Some fixes in the controller manager.

zyxist committed:
Adding the lowest priority for the broker events.

zyxist committed:
Adding the initial facade implementation.

zyxist committed:
Adding facade test files to the application.

zyxist committed:
Fixing the facade service file.

zyxist committed:
Fixing typo in the action controller.

zyxist committed:
Fixing the core exception reference.

zyxist commented:
Actually, there are hard to merge, because you've mixed different changes of different components in one commit. The other commits are wrong and break many things:
- Layout changes - completely wrong. You've introduced redundant methods (`addView()`) and responsibilities this class should not even be aware of (script/style management).
- Router changes - no, I'm not crazy. This feature you claim "crazy" is actually quite useful, but needs some further development, not disabling it.
- Multiple templates in a view - this is not what I was thinking of and what I was explaining to you.

megawebmaster commented:
1. Layout method addView() is more useful than you think, i.e. when you want to replace a view for some placeholders.
2. In my opinion it is very unuseful, because adds some data to created urls, which I should be a master of. This provides many problems while I'm telling router to assemble an url and the router adds i.e. a title of current post to url to blog.
3. You said that multiple templates are useful too, but it's your project. I have to save my patches and add them to every commit you will pull. There are a place for view chaining, but this needs to change some thing which is bad imo. I'll keep looking for solution of this problem.

zyxist committed:
Trinity web view and view broker interface updated.

zyxist committed:
Updating the test application to reflect the view changes.

zyxist committed:
Implementing kept routed variables in the router.

zyxist committed:
Updating service interface.

zyxist committed:
Fixing WebUtils service interfaces.

zyxist committed:
Fixing Template service interfaces.

zyxist committed:
Adding module template resource.

megawebmaster committed:
Added \Trinity\Utils\Config\Loader_Yaml. Modified system files to
support it.

megawebmaster committed:
Added shortcut for \Trinity\Web\Facade\Manager.

megawebmaster committed:
Added right trimming unneeded slash and question mark.

megawebmaster committed:
Added shortcut for \Trinity\Template\Layout in \Trinity\Web\View.

megawebmaster committed:
Added \Trinity\Template\Exception, modified \Trinity\Template\Layout -
added support for assigning Opt_View into a variable, slightly modified
placeholders hanging (for assignView() and performance purposes).

megawebmaster committed:
Added Helper_Styles and Helper_Javascripts, modified demonstration
Application to show how to use them.

zyxist commented:
Outdated request.
